test/Fixtures/Fixtures.yaml

```diff
@@ -0,0 +1,902 @@
+    CoordinateReferenceSystem:
+  EmbeddedChecksum:
+    Type:
+    Value:
+  EndDate:
+  BlockAlign: 2
+  Conditionals:
+    FilenameExtractor:
+      Extension: .wav
+      Stem: S4U09523_20210621_212111
+      StartDate:
+      LocalStartDate: 2021-06-21T21:21:11
+      Location:
+
+- Name: Firmware 3.3
+  FixturePath: FL_BAR_LT/3.30_NewFirmware/00008570_Dartmouth_20220331T094900-0300_Rec.flac
+  Notes:
+  Extension: .flac
+  Stem: 00008570_Dartmouth_20220331T094900-0300_Rec
+  ValidMetadata: Yes
+  MimeType: audio/flac
+  FrontierLabsLogFile: FL_BAR_LT/3.30_NewFirmware/logfile_00008570_SD1.txt
+  CanProcess:
+    - FilenameExtractor
+    - FlacHeaderExtractor
+    - FlacCommentExtractor
+    - FrontierLabsLogFileExtractor
+  Process:
+    - FilenameExtractor
+    - FlacHeaderExtractor
+    - FlacCommentExtractor
+    - FrontierLabsLogFileExtractor
+    - FLCommentAndLogExtractor
+  Vendor: Frontier Labs
+  StartDate: 2022-03-31T09:49:02-03:00
+  LocalStartDate: 2022-03-31T09:49:02
+  CalculatedChecksum:
+    Type:
+    Value:
+  DurationSeconds: 31703040/44100
+  TotalSamples: 31703040
+  Channels: 1
+  SampleRateHertz: 44100
+  BitsPerSecond: 705600
+  BitDepth: 16
+  MemoryCard:
+    FormatType: exFAT
+    ManufacturerID: 3
+    OEMID: SD
+    ProductName: SD128
+    ProductRevision: 8.5
+    SerialNumber: 1132114800
+    ManufactureDate: 2020/08
+    Speed: 50
+    Capacity: 124868608
+    WrCurrentVmin: 5
+    WrCurrentVmax: 5
+    WriteBlSize: 512
+    EraseBlSize: 65536
+  FileLengthBytes:
+  Sensor:
+    Name:
+    Firmware: 3.30
+    SerialNumber: 00008570
+    PowerSource: Ext-power
+    Microphones:
+      - Type: STD AUDIO MIC
+        UID: 009966
+        BuildDate: 2019-03-20
+        Gain: 50dB
+        Channel: 2
+        ChannelName: B
+        Sensitivity:
+    Voltage:
+    Gain:
+    Configuration:
+    BatteryLevel: 65p 3.79V
+    Temperature:
+    LastTimeSync: 2022-03-31T09:48:40-03:00
+  Location:
+    Longitude: -65.95084
+    Latitude: 43.70444
+    Altitude:
+    HorizontalAccuracy:
+    VerticalAccuracy:
+    Speed:
+    Course:
+    SampleDateTime:
+    CoordinateReferenceSystem:
+  EmbeddedChecksum:
+    Type:
+    Value:
+  EndDate: 2022-03-31T10:01:02-03:00
+  BlockAlign:	
+  Conditionals:
+    FilenameExtractor:
+      Extension: .flac
+      Stem: 00008570_Dartmouth_20220331T094900-0300_Rec
+      StartDate: 2022-03-31T09:49:00-03:00
+      LocalStartDate: 2022-03-31T09:49:00
+      Location:
+    FrontierLabsLogFileExtractor:
+      MemoryCard:
+        FormatType: exFAT
+        ManufacturerID: 3
+        OEMID: SD
+        ProductName: SD128
+        ProductRevision: 8.5
+        SerialNumber: 8041328
+        ManufactureDate: 2020/08
+        Speed: 50
+        Capacity: 124868608
+        WrCurrentVmin: 5
+        WrCurrentVmax: 5
+        WriteBlSize: 512
+        EraseBlSize: 65536
+      Sensor:
+        Firmware: 3.30
+        SerialNumber: 00008570
+        PowerSource: Ext-power     
```

Comment:
This is the structure I used for different cases in the YAML file. Conditionals is storing completely separate FixtureModel objects, to be used in specified cases (with specific extractors). Tests are now passing.

This wasn't the set up I was hoping to implement. I was hoping the conditionals could store only the values that needed to be changed when a different extractor is used, and that I could then apply those new values on to my existing FixtureModel, rather than creating a new one from scratch. Though I've tried different strategies, I haven't been able to get this to cooperate for a variety of reasons.

Figured I'd check in with what I have to see you think, or if you had any better ideas on how to configure the YAML for these cases.

Reply:
I need to think about this more. 

I think we could probably merge conditionals into the `Process` field. Each process entry can be a name with an object. Empty object have no overrides, non-empty override specific fields. 

e.g.
[
```yaml
  Process:
    # override nothing
    - FilenameExtractor: {}
    # Override SD card serial number
    - FlacHeaderExtractor: 
        MemoryCards: 
            - SerialNumber: banana
```

I'm not sure on this though, so we'll discuss tomorrow.